### PR TITLE
#60686 Remove priority concept from workflows and states and review sidekiq queues

### PR DESCRIPTION
**Pull request by serch**
@npepinpe could you please take a look?

I removed the concept of priority from both the workflows and the states
I linted much of the code
and I removed the cim:release task, as this is not a gem we can just use `rake release`

The corresponding PR in BarcooBackend is #214 but I'm still working on it

### CODE REVIEW

File: lib/rworkflow/flow.rb [hunk 1]
```diff
@@ -319,7 +318,7 @@ def start(objects)
     end
 
     def total_objects_processed(counters = nil)
-      return (counters || get_counters).reduce(0) do |sum, pair|
+      return (counters || counters).reduce(0) do |sum, pair|
```
Comment (npepinpe) [inline lib/rworkflow/flow.rb]: does that even work? the local variable probably shadows the method

File: lib/rworkflow/flow.rb [hunk 2]
```diff
@@ -406,12 +405,14 @@ def load(id, klass = nil)
       def read_flow_class(id)
         klass = nil
         raw_class = id.split('__').first
-        klass = begin
-          raw_class.constantize
-        rescue NameError => _
-          Rails.logger.warn("Unknown flow class for workflow id #{id}")
-          nil
-        end if !raw_class.nil?
+        if !raw_class.nil?
```
Comment (npepinpe) [inline lib/rworkflow/flow.rb:408]: unless raw_class.nil? or if raw_class.present? please

File: lib/rworkflow/lifecycle.rb [hunk 3]
```diff
@@ -7,9 +7,8 @@ class Lifecycle
 
     DEFAULT_STATE_OPTIONS = {
       cardinality: State::DEFAULT_CARDINALITY,
-      priority: State::DEFAULT_PRIORITY,
       policy: State::STATE_POLICY_NO_WAIT
-    }
+    }.freeze
 
     def initialize(state_class: State, state_options: {})
       @state_options = DEFAULT_STATE_OPTIONS.merge(state_options)
```
Comment (npepinpe) [inline lib/rworkflow/lifecycle.rb]: should really be self.class::DEFAULT_STATE_OPTIONS to allow subclasses to overload the constant

File: lib/rworkflow/sidekiq_helper.rb [hunk 4]
```diff
@@ -3,35 +3,14 @@
 module Rworkflow
   module SidekiqHelper
     def self.included(klass)
-      klass.send :extend,  ClassMethods
-      klass.send :include, InstanceMethods
+      klass.send :extend, ClassMethods
```
Comment (npepinpe) [inline lib/rworkflow/sidekiq_helper.rb:6]: if we have a hard dependency on Rails, you can just use ActiveSupport::Concern no?

**Review by npepinpe — CHANGES_REQUESTED** [review]

File: lib/rworkflow/sidekiq_helper.rb (hunk 4 above)
Comment (serch) [reply lib/rworkflow/sidekiq_helper.rb:6]: I'd say let's not make it more dependent on rails until we need to make it dependent on rails

**Pull request merged**